python/tk_hiero_export/sg_shot_processor.py

```diff
@@ -256,6 +194,73 @@ def _get_tags(self, items):
         tags = [tag for tag in tags if "Nuke Project File" not in tag.name()]
         return tags
 
+class ShotgunShotProcessor(ShotgunHieroObjectBase, FnShotProcessor.ShotProcessor):
+    """
+    Adds hook functionality to the built in Shot processor.
+    """
+    def __init__(self, preset, submission=None, synchronous=False):
+        FnShotProcessor.ShotProcessor.__init__(self, preset, submission, synchronous)
+
+        # Call pre processor hook here to make sure it happens pior to any 'hook_resolve_custom_strings'.
+        # The order if execution is basically [init processor, resolve user entries, startProcessing].
+        self.app.execute_hook("hook_pre_export", processor=self)
```

Comment:
If we are passing this (internal?) object into the hook, are we at risk breaking clients' hooks now due to the refactors? Not related to the current changes, but should be worried about leaky abstraction here when we are passing the `ShotgunShotProcessor` instance down into the hook? Curious about that design decision. Normally, when we do hooks, we try to pass simple data types across the hook boundary, dicts, strings, ints etc.